src/Compilers/CSharp/Portable/Parser/SlidingTextWindow.cs

```diff
@@ -34,207 +45,165 @@ internal sealed class SlidingTextWindow : IDisposable
         /// </summary>
         public const char InvalidCharacter = char.MaxValue;
 
-        private const int DefaultWindowLength = 2048;
+        /// <summary>
+        /// This number was picked based on the roslyn codebase.  If we look at parsing all files,
+        /// this window ensures that 97.4% of all characters grabbed are within the window.  Note
+        /// that this is lower than expected, but heavily influenced by how heavily roslyn uses
+        /// large strings in its codebase to represent test content.  With all these files, lexemes
+        /// average around 60 characters.  If tests are excluded, then average lexeme length drops
+        /// to 25 characters, and the hit rate goes up to 98.4%.  Because this hitrate is so high,
+        /// we don't bother to dynamically resize the window.  Instead, when we hit the end, we just
+        /// grab in the next chunk of characters from the source text, as only one in every 65 lexemes
+        /// will need to go back to the source-text to read the full lexeme.
+        /// </summary>
+        public const int DefaultWindowLength = 4096;
 
-        private readonly SourceText _text;                 // Source of text to parse.
-        private int _basis;                                // Offset of the window relative to the SourceText start.
-        private int _offset;                               // Offset from the start of the window.
-        private readonly int _textEnd;                     // Absolute end position
-        private char[] _characterWindow;                   // Moveable window of chars from source text
-        private int _characterWindowCount;                 // # of valid characters in chars buffer
```

Comment:
primary goal was to simplify the heck out of this goop.  Instead of having lots of things to keep in sync, we simplify our lives by using well known types. Specifically, ArraySegment encapsulates the window and its count.    And there's no need for a separate offset/basis.